File: 2021/1/javascript/main.js
const readline = require('readline');
const fs = require('fs');

let previous = null;
let counter1 = 0;

let a = null;
let b = null;
let c = null;
let counter2 = 0;

const readInterface = readline.createInterface({
    input: fs.createReadStream('../input'),
});

readInterface.on('line', function (line) {
    const parsedInt = parseInt(line, 10);
    if (previous != null && previous < parsedInt) {
        counter1++;
    }
    previous = parsedInt;

    if (a != null &&
        b != null &&
        c != null &&
        (a + b + c < b + c + parsedInt)) {
        counter2++;
    }
    a = b;
    b = c;
    c = parsedInt;
}).on('close', function (line) {
    console.log("Aufagbe 1: " + counter1);
    console.log("Aufgabe 2: " + counter2);
});

File: 2021/1/dart/main.dart
import 'dart:io';
import 'dart:convert';

Future<void> main(List<String> args) async {
  int? previous = null;
  int counter1 = 0;

  int? a = null;
  int? b = null;
  int? c = null;
  int counter2 = 0;

  Stream<String> lines = new File("input")
      .openRead()
      .map(utf8.decode)
      .transform(new LineSplitter());

  await for (var line in lines) {
    int parsedInt = int.parse(line);
    if (previous != null && previous < parsedInt) {
      counter1++;
    }
    previous = parsedInt;

    if (a != null &&
        b != null &&
        c != null &&
        (a + b + c < b + c + parsedInt)) {
      counter2++;
    }
    a = b;
    b = c;
    c = parsedInt;
  }

  print("Aufagbe 1: $counter1");
  print("Aufgabe 2: $counter2");
}
